File: ReSharper.FSharp/src/FSharp.Psi.Features/src/ZoneMarker.fs
namespace JetBrains.ReSharper.Plugins.FSharp

open JetBrains.Application.BuildScript.Application.Zones
open JetBrains.ReSharper.Plugins.FSharp

[<ZoneMarker>]
type ZoneMarker() =
    interface IRequire<ILanguageFSharpZone>

File: ReSharper.FSharp/src/FSharp.Fantomas.Protocol/src/ZoneMarker.cs
using JetBrains.Application.BuildScript.Application.Zones;
using JetBrains.ProjectModel;
using JetBrains.Rider.Model;

namespace JetBrains.ReSharper.Plugins.FSharp.Fantomas.Protocol
{
  [ZoneMarker]
  public class ZoneMarker : IRequire<IProjectModelZone>, IRequire<IRiderModelZone>
  {
  }
}
